src/plugins/d3/renderer/hooks/useLegend/index.ts

```diff
@@ -1,18 +1,34 @@
 import React from 'react';
 
-import type {ChartKitWidgetSeries} from '../../../../../types/widget-data';
+import type {ChartKitWidgetSeries, PieSeriesData} from '../../../../../types/widget-data';
 
-import {getVisibleSeriesNames} from '../../utils';
+import {getVisibleEntriesNames, isAxisRelatedSeries} from '../../utils';
 
 export type OnLegendItemClick = (data: {name: string; metaKey: boolean}) => void;
 
 type Args = {
     series: ChartKitWidgetSeries[];
 };
 
+const getActiveLegendItems = (series: ChartKitWidgetSeries[]) => {
+    if (series.length === 1 && !isAxisRelatedSeries(series[0])) {
```

Comment:
I don't think it's appropriate relying on such a condition.
The situation when two series with different types are passed to chart seems quite valid